frontend-v2/src/features/trial/Doses.tsx

```diff
@@ -130,13 +134,66 @@ const Doses: FC<Props> = ({ onChange, project, protocol, units }) => {
     removeDose(index);
   };
 
+  const selectedAmountId = getValues("amount_unit");
+  const selectedAmountLabel =
+    baseUnit?.compatible_units?.find(({ id }) => id === selectedAmountId)
```

Comment:
There's a TypeScript error that breaks the build. I think it might be comparing a string with a number. I've been running into a few of these errors in the Results page work. My usual solution is to coerce a string to a number with `+`:

```ts
const foundThing = SomeArray.find(numberId => numberId === +someStringId)
```

Reply:
Fixed